sql/testdata/builtin_function

```diff
@@ -332,3 +332,286 @@ SELECT extract(nanosecond from timestamp '2001-04-10 12:04:59.34565423')
 
 query error extract: unsupported timespan: nansecond
 SELECT extract(nansecond from timestamp '2001-04-10 12:04:59.34565423')
+
+query T
+SELECT abs(-1.0)
+----
+1
+
+query T
+SELECT abs(-0.0)
+----
+0
+
+query T
+SELECT abs(1.0)
+----
+1
+
+query error abs: abs of min integer value \(-9223372036854775808\) not defined
+SELECT abs(-9223372036854775808)
+
+query T
+SELECT abs(-9223372036854775807)
+----
+9223372036854775807
+
+query T
+SELECT abs(0)
+----
+0
+
+query T
+SELECT abs(1)
+----
+1
+
+query T
+SELECT abs(sin(pi())) < 1e-12
+----
+true
+
+query T
+SELECT acos(0.5)
+----
+1.0471975511965976
+
+query T
+SELECT asin(0.5)
+----
+0.5235987755982989
+
+query T
+SELECT asin(1.5)
+----
+NaN
+
+query T
+SELECT atan(0.5)
+----
+0.4636476090008061
+
+query T
+SELECT atan2(10.0, 5.0)
+----
+1.1071487177940904
+
+query T
+SELECT ceil(0.5)
+----
+1
+
+query T
+SELECT ceiling(0.5)
+----
+1
+
+query T
+SELECT cos(0.5)
+----
+0.8775825618903728
+
+query T
+SELECT degrees(0.5)
+----
+28.64788975654116
+
+query T
+SELECT div(1.0, 2.0)
+----
+0.5
+
+query T
+SELECT div(1.0, 0.0)
+----
++Inf
+
+query T
+SELECT exp(1.0)
+----
+2.718281828459045
+
+query T
+SELECT floor(1.5)
+----
+1
+
+query T
+SELECT ln(2.0)
+----
+0.6931471805599453
+
+query T
+SELECT log(10.0)
+----
+1
+
+query T
+SELECT mod(5.0, 2.0)
+----
+1
+
+query T
+SELECT mod(1.0, 0.0)
+----
+NaN
+
+query T
+SELECT mod(5, 2)
+----
+1
+
+# mod returns the same results as PostgreSQL 9.4.4
+# in tests below (except for the error message).
+
+query error mod: division by zero
+SELECT mod(5, 0)
+
+query T
+SELECT mod(-100, 8)
+----
+-4
+
+query T
+SELECT mod(-100, -8)
+----
+-4
+
+query T
+SELECT mod(-9223372036854775808, 3)
+----
+-2
+
+query T
+SELECT mod(-9223372036854775808, -1)
+----
+0
+
+query T
+SELECT pi()
+----
+3.141592653589793
+
+query T
+SELECT pow(2.0, 3.0)
+----
+8
+
+query T
+SELECT power(2.0, 3.0)
+----
+8
+
+query T
+SELECT radians(45.0)
+----
+0.7853981633974483
+
+query error number of digits should be greater or equal 0
+SELECT round(1.0, -1)
+
+query T
+SELECT round(1.23, 0)
+----
+1
+
+query T
+SELECT round(1.5, 0)
+----
+2
+
+# round to nearest even
+query T
+SELECT round(2.5, 0)
+----
+2
+
+query T
+SELECT round(1.12, 1)
+----
+1.1
+
+query T
+SELECT round(1.1234, 3)
+----
+1.123
+
+query T
+SELECT sign(-2)
+----
+-1
+
+query T
+SELECT sign(0)
+----
+1
+
+query T
+SELECT sign(2)
+----
+1
+
+query T
+SELECT sign(-2.0)
+----
+-1
+
+query T
+SELECT sign(-0.0)
+----
+-1
+
+query T
+SELECT sign(0.0)
+----
+1
+
+query T
+SELECT sign(2.0)
+----
+1
+
+query T
+SELECT sin(0.0)
+----
+0
+
+query T
+SELECT sin(1.0)
+----
+0.8414709848078965
+
+query T
+SELECT sqrt(4.0)
+----
+2
+
+query T
+SELECT sqrt(-1.0)
+----
+NaN
+
+query T
+SELECT tan(0.0)
+----
+0
+
+query T
+SELECT tan(5.0)
+----
+-3.3805150062465854
+
+query T
+SELECT trunc(0.0)
+----
+0
+
+query T
+SELECT trunc(-0.0)
+----
+-0
+
+query T
+SELECT trunc(1.9)
```

Comment:
At some point we're just testing the behavior of the routines we're calling in the `math` package, which isn't terribly interesting. I think a single test case per input type and boundary condition in our implementation (not the `math` implementation) is sufficient.